01_git/scissors/main.c

```diff
@@ -1 +1,54 @@
 
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+int main()
+{
+    enum {
+        ROCK,
+        PAPER,
+        SCISSORS,
+        VARS_NUM
+    } vars;
+    char input[16];
+    int player, ai;
+    const char* option[] = {"rock", "paper", "scissors"};
+
+    while(1) {
+        
+        printf("Please choose: rock (r) - paper (p) - scissors (s) and press \"Enter\"\n");
+        
+        fgets(input, sizeof(input), stdin);
+        
+        switch((char)input[0]) {
+            case 'r':
+                player = ROCK;
+                break;
+            case 'p':
+                player = PAPER;
+                break;
+            case 's':
+                player = SCISSORS;
+                break;
+            default:
+                printf("WRONG INPUT!!!\n\n");
+                continue;
+        }
+        
+        ai = rand() % VARS_NUM;
+        printf("You choose %s, I choose %s\n", option[player], option[ai]);
+        
+        if(player == ai) {
+            printf("Draw... Try again!\n\n");
+            continue;
```

Comment:
According to that logic you force the player to play till someone wins. Other hands when it happened game close without any chance to continue.